appveyor.yml

```diff
@@ -0,0 +1,74 @@
+version: '{branch}.{build}'
+branches:
+  only:
+  - master
+
+image: Visual Studio 2017
+
+environment:
+  PHP_SDK_BINARY_TOOLS_VER: php-sdk-2.0.9
+
+  matrix:
+    - ARCH: x86
+      PHP_VER: 7.2
+      VC_VER: vc15
+    - ARCH: x64
+      PHP_VER: 7.2
+      VC_VER: vc15
+install:
+- cmd: cinst wget
+build_script:
+- cmd: >-
+    "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" %ARCH%
+
+    wget https://github.com/OSTC/php-sdk-binary-tools/archive/%PHP_SDK_BINARY_TOOLS_VER%.zip --no-check-certificate -q -O php-sdk-binary-tools-%PHP_SDK_BINARY_TOOLS_VER%.zip
+
+    7z x -y php-sdk-binary-tools-%PHP_SDK_BINARY_TOOLS_VER%.zip -oC:\projects
+
+    move C:\projects\php-sdk-binary-tools-%PHP_SDK_BINARY_TOOLS_VER% C:\projects\php-sdk
+    
+    C:\projects\php-sdk\bin\phpsdk_setvars.bat
+
+    git clone https://github.com/php/php-src C:\projects\php-src -b PHP-%PHP_VER% --depth=1
+
+    mkdir C:\projects\php-src\ext\ast
+
+    xcopy C:\projects\php-ast C:\projects\php-src\ext\ast /s /e /y /q
+
+    REM FIXME no PHP 7.2 deps downloads available yet - wget http://windows.php.net/downloads/php-sdk/deps-%PHP_VER%-%VC_VER%-%ARCH%.7z -q
+    
+    wget http://windows.php.net/downloads/php-sdk/deps-master-%VC_VER%-%ARCH%.7z -q

```

Comment:
I'm getting a build failure here (https://ci.appveyor.com/project/nikic/php-ast/build/master.5/job/67xwu68f2x4iopmb) because the master deps have been dropped. What is one supposed to use now? /cc @weltling

Reply:
https://github.com/phan/phan/blob/master/appveyor.yml is another project that builds php-ast

I'd suggest cloning PHP-7.1 instead of master from php-src, as well as the dependencies. The command would be:

`git clone --depth=1 --branch=PHP-7.1 https://github.com/php/php-src C:\projects\php-src`

(There's other lines that would need to be changed, see the other appveyor.yml

Reply:
Looking at https://wiki.php.net/internals/windows/stepbystepbuild_sdk_2#download_prerequisites calling `phpsdk_deps -u` instead of manually downloading the deps should fix this.

Reply:
@nikic yeah, `phpsdk_deps -u` and caching the result is what we do for the core, too. But i also guess, you don't need to do it at all, as `php-ast` doesn't require any dependencies except the core.

Perhaps you'd also find this useful https://github.com/weltling/parle/blob/master/.appveyor.yml. I used another approach there, the phpize way, which of course runs much faster than building the whole PHP.  It might be not perfect enough yet to present as a universal solution. For exts depending on non core libs or exts it'd be more complicated. Anyway using phpize would make a good speedup for the builds.

Thanks.

Reply:
Using phpsdk_deps seems to be working after a few tries, but is incredibly slow (20 minutes into the [build](https://ci.appveyor.com/project/nikic/php-ast/build/job/dd8wuhpqkm888urc) and still going, is that normal?), so I guess this isn't an option. I agree that using phpize is much better, but the configuration for that looks even more complicated :(

Reply:
Could be that appveyor is just terribly slow at the moment. I've noticed this before on Sundays.

Reply:
It's not going, from what i see. The call to `C:\projects\php-sdk\phpsdk-%VC_VER%-%ARCH%.bat` starts the SDK shell and it hangs, no further commands are executed. It is normal, because this commands starts a new CMD shell, say as if i would call `bash` while already on a bash shell. The SDK shell also creates the necessary environment for the internal tools. 

Instead the commands need to be packed in a script that will run as a task. Please check like here https://github.com/php/php-src/blob/master/appveyor/build.bat#L32 . Like `phpsdk-vc14-x64.bat -t my_build_script.bat`, then `my_build_script.bat` will be run with all the environment necessary. In the previous link, i create that task script dynamically https://github.com/weltling/parle/blob/master/.appveyor.yml#L89 depending on the build matrix.

Yeah, i'm not a big powershell specialist, too :) Pehraps this appveyor.yml should be revamped, either for phpize or in any case to use this task scheme. I'll see to come to it later today or in the near future, if no one beats me to it.

Thanks.

Reply:
@weltling Ahhhh, yeah, you're right. I switched to using the starter script because phpsdk_deps requires the PHP_VC_VER env variable which seems to be set by it, but I did not realize that this starts a new shell. I've just disabled AppVeyor for now -- it's hard enough for me to get Travis to work most of the time, I don't think I stand a chance here ^^

Reply:
@nikic As php-ast does not need any deps, you could just have removed the lines regarding the deps. Proof of the pudding:
https://github.com/Jan-E/php-ast/commits/appveyor